tests/app/helpers.py

```diff
@@ -89,19 +92,24 @@ def services():
             ]
         }
 
+    def teardown_login(self):
+        if getattr(self, 'original_get_user', None) is not None:
+            data_api_client.get_user = self.original_get_user
+            self.original_get_user = None
+
     def login(self):
-        data_api_client.authenticate_user = Mock(
-            return_value=(self.user(
-                123, "email@email.com", 1234, 'Supplier Name', 'Name')))
+        with patch('app.main.views.login.data_api_client') as login_api_client:
+            login_api_client.authenticate_user.return_value = self.user(
+                123, "email@email.com", 1234, 'Supplier Name', 'Name')
 
-        data_api_client.get_user = Mock(
-            return_value=(self.user(
-                123, "email@email.com", 1234, 'Supplier Name', 'Name')))
+            self.original_get_user = data_api_client.get_user
```

Comment:
Instead of saving the old method we could use `patch` here manually:
```
self.patch = mock.patch(data_api_client, 'get_user', ...)
self.patch.start()
...
self.patch.stop()
```